src/components/main/process/hooks/useProcessorUpdateOpt.ts

```diff
@@ -0,0 +1,382 @@
+import { useCallback, useContext, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { LogAllow } from "@_constants/main";
+import {
+  parseFile,
+  serializeFile,
+  TFileNodeData,
+  writeFile,
+} from "@_node/file";
+import {
+  parseHtmlCodePart,
+  serializeHtml,
+  THtmlNodeData,
+  THtmlPageSettings,
+} from "@_node/html";
+import { TNode, TNodeTreeData, TNodeUid } from "@_node/types";
+import {
+  fnSelector,
+  MainContext,
+  navigatorSelector,
+  setCurrentFileContent,
+} from "@_redux/main";
+import { TFileInfo, TFileType } from "@_types/main";
+import {
+  addNewNode,
+  detectSeedNodeChanges,
+  generateFileInfo,
+  getChangedUids,
+  getHtmlPageSettings,
+  getPreViewPath,
+  refreshIframeIfSeedNodeChanges,
+  removeOrgNode,
+  replaceElementInIframe,
+} from "../helpers";
+
+export const useProcessorUpdateOpt = () => {
+  const dispatch = useDispatch();
+  const { file } = useSelector(navigatorSelector);
+  const { focusedItem } = useSelector(fnSelector);
+  const {
+    // global action
+    addRunningActions,
+    removeRunningActions,
+    // file tree view
+    setFSPending,
+    ffTree,
+    setFFNode,
+    parseFileFlag,
+    currentFileUid,
+    // node tree view
+    nodeTree,
+    setNodeTree,
+    nodeMaxUid,
+    setNodeMaxUid,
+    // stage view
+    setIFrameSrc,
+    fileInfo,
+    setFileInfo,
+    setNeedToReloadIFrame,
+    // code view
+    setCodeEditing,
+    codeChanges,
+    setNewFocusedNodeUid,
+    // processor
+    updateOpt,
+    setUpdateOpt,
+    // references
+    htmlReferenceData,
+    // other
+    osType,
+  } = useContext(MainContext);
+  // -------------------------------------------------------------- sync --------------------------------------------------------------
+
+  // service - get reference data for current file type
+  const getReferenceData = useCallback(
+    (fileType: TFileType) => {
+      return fileType === "html" ? htmlReferenceData : htmlReferenceData;
+    },
+    [htmlReferenceData],
+  );
+
+  useEffect(() => {
+    if (updateOpt.parse === true) {
+      let onlyRenderViewState = false;
+      // parse file content
+      let _nodeTree: TNodeTreeData = JSON.parse(JSON.stringify(nodeTree));
```

Comment:
@VictoriaShyika is there something we can do to breakdown the code inside this useEffect into meaningful functions without compromising with the functionalities.

Reply:
added helpers functions to each block, in this way, the functionality works correctly